dll/win32/shell32/CDefView.cpp

```diff
@@ -3435,7 +3435,7 @@ HRESULT WINAPI CDefView::GetItemObject(UINT uItem, REFIID riid, LPVOID *ppvOut)
 FOLDERVIEWMODE CDefView::GetDefaultViewMode()
 {
     FOLDERVIEWMODE mode = ((m_FolderSettings.fFlags & FWF_DESKTOP) || !IsOS(OS_SERVERADMINUI)) ? FVM_ICON : FVM_DETAILS;
-    FOLDERVIEWMODE temp;
+    FOLDERVIEWMODE temp = mode;
```

Comment:
Is this necessary? (I would expect `temp` gets initialized to something only when the DoFolderViewCB returns a success code, in which case, the rest of the if-tests should use an initialized `temp`, and the `mode = temp` would also use an initialized value.)

Reply:
`S_FALSE` is a success code but might not update `temp`

Reply:
ah, good point indeed, it's always a bit unintuitive.